data-transform/src/main/java/com/linkedin/data/transform/filter/request/MaskTree.java

```diff
@@ -47,14 +44,14 @@
  * The semantics of applying a MaskTree to a DataMap are documented in {@link com.linkedin.data.transform.DataComplexProcessor}
  */
 
-public class MaskTree
+public class MaskTree extends FieldMask
```

Comment:
MaskMap was my first choice :)  and renamed it to FieldMask as this is actually storing mask for the field tree. 
I see your point that this can represent any generic mask. I can rename it back.